lib/common/grpc-service.js

```diff
@@ -0,0 +1,337 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module common/grpcService
+ */
+
+'use strict';
+
+var camelize = require('camelize');
+var googleProtoFiles = require('google-proto-files');
+var grpc = require('grpc');
+var is = require('is');
+var nodeutil = require('util');
+var path = require('path');
+var snakeize = require('snakeize');
+
+/**
+ * @type {module:common/service}
+ * @private
+ */
+var Service = require('./service.js');
+
+/**
+ * @const {object} - A map of protobuf codes to HTTP status codes.
+ * @private
+ */
+var HTTP_ERROR_CODE_MAP = {
```

Comment:
I keyed from the code of the gRPC code. It would be a kind of trick to use this as an array. I prefer the explicit style, but can go either way.